backends/cuda-ref/ceed-cuda-ref-operator.c

```diff
@@ -1557,6 +1635,17 @@ static int CeedOperatorLinearAssembleAddDiagonalAtPoints_Cuda(CeedOperator op, C
   max_num_points = impl->max_num_points;
   for (CeedInt i = 0; i < num_elem; i++) num_points[i] = max_num_points;
 
+  // Create separate output e-vecs
+  if (impl->has_shared_e_vecs) {
```

Comment:
I don't know the normal call structure for this, but are the shared e vecs re-setup when another OperatorApply is done? I'm thinking in non-linear cases where we continuously build the diagonal throughout the simulation.

Reply:
Nope, for this case once we swap away from shared input/output e-vecs, then that's the state the operator stays in. This only effects AtPoints and hopefully we'll be moving away from actually computing the diagonal for the AtPoints operator itself but instead working off an operator using the background mesh quadrature, so this shouldn't be a big issue in the future